packages/next-app/src/hooks/useTransaction.ts

```diff
@@ -190,3 +222,109 @@ export function useMetadataTransaction({
     transactionUrl,
   };
 }
+
+export function useInternalBalancesTransaction({
+  userAddress,
+  token,
+}: {
+  userAddress: `0x${string}`;
+  //TODO change to useInternalBalances when subgraph is updated
+  //https://linear.app/bleu-llc/issue/BAL-272/fix-internal-balance-token-attribute-on-subgraph
+  token: any;
+}) {
+  const { chain } = useNetwork();
+  const [isNotifierOpen, setIsNotifierOpen] = useState(false);
+  const [notification, setNotification] = useState<Notification | null>(null);
+  const [transactionUrl, setTransactionUrl] = useState<string>();
+  const [operationKind, setOperationKind] = useState<UserBalanceOpKind>();
+
+  //Prepare data for transaction
+  const userBalanceOp = {
+    kind: operationKind as number,
+    asset: token.token as `0x${string}`,
+    amount: parseFixed(token.balance, 18),
+    sender: userAddress as `0x${string}`,
+    recipient: userAddress as `0x${string}`,
+  };
+
+  const { config } = usePrepareVaultManageUserBalance({
+    args: [[userBalanceOp]],
+  });
+
+  const { data, write } = useVaultManageUserBalance(config);
+
+  //trigger transaction
+  function handleWithdraw() {
+    setOperationKind(UserBalanceOpKind.WITHDRAW_INTERNAL);
+    setNotification(
+      NOTIFICATION_MAP_INTERNAL_BALANCES[TransactionStatus.WAITING_APPROVAL]
+    );
+  }
+
+  //trigger the actual transaction
+  useEffect(() => {
+    if (!operationKind) return;
+    write?.();
+  }, [operationKind]);
```

Comment:
Shouldn't this be inside `handleWithdraw`? Why trigger `write` magically here?

Reply:
Will solve this [here](https://linear.app/bleu-llc/issue/BAL-249/implement-internal-balance-withdraw-flow)